From 2ef7a579f29a5c738cf853c3f3dcc3bd7594aa88 Mon Sep 17 00:00:00 2001
From: Jianzheng Zhou <jianzheng.zhou@freescale.com>
Date: Fri, 23 May 2014 19:30:52 +0800
Subject: [PATCH 54/95] =?UTF-8?q?ENGR00315051=C2=A0wfd:fix=201080P=20freeze=20?=
 =?UTF-8?q?in=20wifi=20diplay?=
MIME-Version: 1.0
Content-Type: text/plain; charset=UTF-8
Content-Transfer-Encoding: 8bit

Change to use framerate not over 30fps. and restrict source
encoder to max 720P although it will be 1080P or even more bigger.

Signed-off-by: Jianzheng Zhou <jianzheng.zhou@freescale.com>
---
 media/libstagefright/wifi-display/VideoFormats.cpp | 8 +++++++-
 1 file changed, 7 insertions(+), 1 deletion(-)

diff --git a/media/libstagefright/wifi-display/VideoFormats.cpp b/media/libstagefright/wifi-display/VideoFormats.cpp
index 3acee1d..7f96196 100644
--- a/media/libstagefright/wifi-display/VideoFormats.cpp
+++ b/media/libstagefright/wifi-display/VideoFormats.cpp
@@ -156,9 +156,15 @@ void VideoFormats::setNativeResolution(ResolutionType type, size_t index) {
 void VideoFormats::ConvertDpyInfo2Resolution(DisplayInfo Dpy, ResolutionType& type, size_t& index) {
      for (size_t i = 0; i < kNumResolutionTypes; ++i) {
         for (size_t j = 0; j < 32; ++j) {
+            if (Dpy.w > 1280) {
+                type = RESOLUTION_CEA;
+                index = 5;
+                return;
+            }
+            //fps should be less 30, or performance will downgrade.
             if (mResolutionTable[i][j].width == Dpy.w
                     && mResolutionTable[i][j].height == Dpy.h
-                    && mResolutionTable[i][j].framesPerSecond == (size_t)Dpy.fps) {
+                    && mResolutionTable[i][j].framesPerSecond <= 30) {
                 type = (ResolutionType)i;
                 index = j;
                 return;
-- 
1.8.0

